tailcfg/tailcfg.go

```diff
@@ -2764,3 +2773,40 @@ type EarlyNoise struct {
 // For some request types, the header may have multiple values. (e.g. OldNodeKey
 // vs NodeKey)
 const LBHeader = "Ts-Lb"
+
+// KubernetesCapRule is a rule provided via PeerCapabilityKubernetes capability.
+type KubernetesCapRule struct {
```

Comment:
maybe we should add a k8s specific package for these things?

cc @bradfitz for thoughts

Reply:
k8s-specific package for k8s things that are shared with control and that still requires requires control code owners review? what would that give, just to keep things logically separate?

Reply:
@maisem moved this to ./kube (that does not pull in any kube libraries so should be fine hopefully) https://github.com/tailscale/tailscale/pull/12236/commits/33501215ce6298de66df87af97d876409648fe6a#diff-252cb93f3dc9ac390d9ba50510ffbd8aaaa08f45f4338b4081ea9e3e1b5608fc